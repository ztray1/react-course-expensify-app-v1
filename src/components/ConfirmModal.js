import React from "react";

import Modal from "react-modal";

const ConfirmModal =(props)=>(
    <Modal
      isOpen={!!props.selectedOption}
      onRequestClose={props.handleCancelOption}
      contentLabel="selected Option"
      className="modal"
      appElement={document.getElementById('app')}>
        <h3>Are you sure to remove?</h3>
        <button className="button button--modal" onClick={props.handleConfirmRemoveOption}>Confirm</button>
        <button className="button button--modal" onClick={props.handleCancelOption}>cancel</button>
    </Modal>
)

export default ConfirmModal;