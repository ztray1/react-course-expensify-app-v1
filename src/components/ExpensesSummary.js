import React from "react";
import {connect} from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total"

export const ExpensesSummary=({expenseCount,expenseTotal})=>{
    const expenseWord=expenseCount===1?"expense":"expenses";
    const formattedExpensesTotal=numeral(expenseTotal/100).format("$0,0.00");
    return(
        <div>
            <h1>Viewing {expenseCount} {expenseWord} expenses totalling {formattedExpensesTotal}</h1>
        </div>
    )
}

const mapStatetProps=(state)=>{
    const visibleExpenses=selectExpenses(state.expenses,state.filters);
    return {
        expenseCount:visibleExpenses.length,
        expenseTotal:selectExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStatetProps)(ExpensesSummary);