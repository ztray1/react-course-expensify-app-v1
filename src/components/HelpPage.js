import React from "react";

const HelpPage=()=>(
    <div>
        This is from HelpPage
    </div>
);
export default HelpPage;