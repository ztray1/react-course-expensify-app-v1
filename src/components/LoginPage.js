import React from "react";
import {connect} from "react-redux";;
import {startGithubLogin,startGoogleLogin, startFacebookLogin} from "../actions/auth";

export const LoginPage=({startGithubLogin,startGoogleLogin,startFacebookLogin})=>(
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It is time to get your expenses under control. </p>
            <button className="button" onClick={startGithubLogin}>Login with GitHub</button>
            <button className="button" onClick={startGoogleLogin}>Login with Google</button>
            <button className="button" onClick={startFacebookLogin}>Login with Facebook</button>
        </div>
    </div>
)

const mapDispatchToProps=(dispatch)=>({
    startGithubLogin:()=>dispatch(startGithubLogin()),
    startGoogleLogin:()=>dispatch(startGoogleLogin()),
    startFacebookLogin:()=>dispatch(startFacebookLogin()),
})

export default connect(undefined,mapDispatchToProps)(LoginPage)