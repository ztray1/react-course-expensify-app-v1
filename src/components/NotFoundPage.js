import React from "react";

const notFoundPage=()=>(
    <div>
        404 - <Link to="/">Go Home</Link>
    </div>
);

export default notFoundPage;