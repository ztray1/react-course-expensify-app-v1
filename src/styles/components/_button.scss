.button{
    background:$blue;
    color:white;
    border:none;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding:$s-size;
    margin:0 0 0.5rem 0;
    text-decoration: none;
}

.button--link{
    background:none;
}

.button--secondary{
    background: #888;
}