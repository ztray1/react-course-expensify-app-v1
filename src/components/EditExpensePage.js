import React from "react";
import {connect} from "react-redux";
import ExpenseForm from "./ExpenseForm";
import {startEditExpense} from "../actions/expenses";
import {startRemoveExpense} from "../actions/expenses";
import ConfirmModal from "./ConfirmModal";
import { confirmAlert } from 'react-confirm-alert';
import 'react-confirm-alert/src/react-confirm-alert.css';

export class EditExpensePage extends React.Component{
    state={
        removeConfirm:false
    }
    onSubmit=(expense)=>{
        this.props.startEditExpense(this.props.expense.id,expense)
        this.props.history.push("/")
    }
    /*onRemove=()=>{
        this.setState({removeConfirm:true})
    }*/
    onRemove=()=> {
        confirmAlert({
          title: 'Confirm to submit',
          message: 'Are you sure to do this.',
          buttons: [
            {
              label: 'Yes',
              onClick: () => {
                this.props.startRemoveExpense({id:this.props.expense.id});
                this.props.history.push("/");
            }
            },
            {
              label: 'No',
              onClick: () => {}
            }
          ]
        });
      };
    onHandleCancelOption=()=>{
        this.setState({removeConfirm:false})
    }
    onHandleConfirmRemoveOption=()=>{
        this.props.startRemoveExpense({id:this.props.expense.id});
        this.props.history.push("/");
    }
    render(){
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}/>
                    <button className="button button--secondary" onClick={this.onRemove}>Remove Expense</button>
                </div>
            </div>
        )
    }
   
};

const mapStateToProps=(state,props)=>{
    return {
        expense:state.expenses.find((expense)=>{
            return (expense.id===props.match.params.id);
        })
    }
}

const mapDispatchToProps=(dispatch)=>({
    startEditExpense:(id,expense)=>{dispatch(startEditExpense(id,expense))},
    startRemoveExpense:({id})=>{dispatch(startRemoveExpense({id}))},
})

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);