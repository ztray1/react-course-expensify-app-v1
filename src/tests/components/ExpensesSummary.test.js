 import React from "react";
 import {shallow} from "enzyme";
 import {ExpensesSummary} from "../../components/ExpensesSummary";

 test("should correctly render ExpensesSummary with mutiple expenses",()=>{
    const wrapper=shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot();
 })

 test("should correctlty render ExpensesSummary with mutiple expenses",()=>{
    const wrapper=shallow(<ExpensesSummary expenseCount={23} expensesTotal={235456456456}/>)
    expect(wrapper).toMatchSnapshot();
 })