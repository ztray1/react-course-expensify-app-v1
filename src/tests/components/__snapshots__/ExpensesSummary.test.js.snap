// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctlty render ExpensesSummary with mutiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     expenses totalling 
    $0.00
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with mutiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     expenses totalling 
    $0.00
  </h1>
</div>
`;
